pkg/didcomm/dispatcher/api.go

```diff
@@ -67,3 +70,22 @@ type DIDCommCallback struct {
 
 // Callback type to pass service callbacks.
 type Callback func(DIDCommCallback)
+
+// Event related apis.
+type Event interface {
```

Comment:
Sorry I wasn't clear before about the embedding: I think this new interface (whatever it will be named) should embed the existing `dispatcher.Service` interface, not the other way around. The dispatchers have no use for these new methods.

Reply:
@llorllale Created a [new](https://github.com/hyperledger/aries-framework-go/pull/335/files#diff-6a4454074ff31ccb82bdbcc343255467R93) interface for DIDComm Services.

Reply:
I opened #344 to consider moving this interface to a different package in the future.

Reply:
@llorllale I had [already](https://github.com/hyperledger/aries-framework-go/pull/335#issuecomment-535058106) raised one for that. May be we can close #344